baseapp/abci.go

```diff
@@ -1353,21 +1353,13 @@ func (app *BaseApp) GetBlockRetentionHeight(commitHeight int64) int64 {
 			return y
 		}
 	}
-
 	// Define retentionHeight as the minimum value that satisfies all non-zero
 	// constraints. All blocks below (commitHeight-retentionHeight) are pruned
 	// from CometBFT.
+	// Default is 0
 	var retentionHeight int64
 
-	// Define the number of blocks needed to protect against misbehaving validators
-	// which allows light clients to operate safely. Note, we piggy back of the
-	// evidence parameters instead of computing an estimated number of blocks based
-	// on the unbonding period and block commitment time as the two should be
-	// equivalent.
-	cp := app.GetConsensusParams(app.finalizeBlockState.Context())
-	if cp.Evidence != nil && cp.Evidence.MaxAgeNumBlocks > 0 {
-		retentionHeight = commitHeight - cp.Evidence.MaxAgeNumBlocks
-	}
+	retentionHeight = commitHeight - 1500
```

Comment:
do we have a const for the default snapshot interval we could use here intead of hardcoding 1500? or could we even use that configured value?

Reply:
the configured interval would be checked below then the smaller number would be taken. 

current height: 1_000_000
snapshot Interval: 1000
min retain height: 1 = 1501 with the default

Min retain height = 998_499  Snapshot Interval height = 999_000. We would keep 1501 blocks instead 1k. this works the other direction

Reply:
why do we want to keep 1500? I'm just trying to avoid including a raw number here out of good habit. If its a magic number, then even then we could use a constant. If its not a magic number and has a more explicit reasoning then could we encode that here?

Reply:
1500 comes from this magic number: https://github.com/celestiaorg/celestia-app/blob/7c12ff54581bbe438f58ea022e928c068a62c641/app/default_overrides.go#L290 - the default snapshot interval height. Happy to turn it into a constant. The idea is that all nodes should still keep all the blocks to sync from the latest snapshot to the head of the chain. Maybe it needs to be a slightly larger number